bindings/python/wscript

```diff
@@ -434,44 +522,19 @@ def get_modules_and_headers(bld):
 
 
 
-
-class python_scan_task_collector(Task.TaskBase):
-    """Tasks that waits for the python-scan-* tasks to complete and then signals WAF to exit
-    """
-    after = 'apiscan'
-    before = 'cxx'
-    color = "BLUE"
-    def __init__(self, curdirnode, env, bld):
-        self.bld = bld
-        super(python_scan_task_collector, self).__init__(generator=self)
-        self.curdirnode = curdirnode
-        self.env = env
-
-    def display(self):
-        return 'python-scan-collector\n'
-
-    def run(self):
-        # signal stop (we generated files into the source dir and WAF
-        # can't cope with it, so we have to force the user to restart
-        # WAF)
-        self.bld.producer.stop = 1
-        return 0
-
-
-
 class gen_ns3_compat_pymod_task(Task.Task):
     """Generates a 'ns3.py' compatibility module."""
-    before = 'cxx'
+    before = ['cxxprogram', 'cxxshlib', 'cxxstlib']
     color = 'BLUE'
-    
+
     def run(self):
         assert len(self.outputs) == 1
-        outfile = file(self.outputs[0].abspath(), "w")
+        outfile = open(self.outputs[0].abspath(), "w")
         print("import warnings", file=outfile)
         print('warnings.warn("the ns3 module is a compatibility layer '\
             'and should not be used in newly written code", DeprecationWarning, stacklevel=2)', file=outfile)
         print(file=outfile)
-        for module in self.bld.env['PYTHON_MODULES_BUILT']:
+        for module in ['antenna', 'aodv', 'applications', 'bridge', 'buildings', 'config-store', 'core', 'csma', 'csma-layout', 'dsdv', 'dsr', 'energy', 'fd-net-device', 'flow-monitor', 'internet', 'internet-apps', 'lr-wpan', 'lte', 'mesh', 'mobility', 'netanim', 'network', 'nix-vector-routing', 'olsr', 'point-to-point', 'point-to-point-layout', 'propagation', 'sixlowpan', 'spectrum', 'stats', 'tap-bridge', 'topology-read', 'traffic-control', 'uan', 'virtual-net-device', 'wave', 'wifi', 'wimax','dce']:
```

Comment:
what is this enumeration of ns-3 modules used for?  Again, it is fragile to do it this way; it doesn't account for modules that may be in 'contrib' directory, or modules that may be disabled in the ns-3 build.

Reply:
fa725fff075bd74b4ce706b41e38857aac4efb1c
We never really use this method run, as we depend on the bake/build/lib/python3.8/site-packages/ns3/__init__.py (which is blank). Even if we make our own ```__init__.py```   in bindings, python3.8 would first check this directory and stop here, not discovering other ns3 cpython libraries which are present in another path which is already added to PYTHONPATH. This is why I had to change the PYTHONPATH lookup orderring in this b454fcb7237121816fc45c6ffee0755c2eec9c19